server/app/services/service-service.js

```diff
@@ -55,23 +69,60 @@ async function getServiceById(serviceId) {
  * @param updatedService {Service} Updated service, constructed from the request.
  * @return {Promise<Service>} The updated service
  */
-async function updateService(serviceId, updatedService) {
-
-    const currentService = await Service.findById(serviceId);
+async function updateService(serviceId, updatedService, _embedded) {
+    const currentService = await Service.findById(serviceId, {
+        include: [
+            {
+                model: Category,
+            },
+        ]
+    });
 
-    if (!currentService) throw createUserError('UnknownService', 'This service does not exist');
+    if (!currentService) throw createUserError('UnknownService', 'This Service does not exist');
 
     logger.verbose('Service service: updating service named %s', currentService.name);
 
-    return await currentService.update({
-        name: updatedService.name,
-        startAt: updatedService.startAt,
-        endAt: updatedService.endAt,
-        nbMax: updatedService.nbMax,
-        category: updatedService.category
-    });
+    const transaction = await sequelize.transaction();
+
+    try {
+        await currentService.update(cleanObject({
+            name: updatedService.name,
+            startAt: updatedService.startAt,
+            endAt: updatedService.endAt,
+            nbMax: updatedService.nbMax,
+            category: updatedService.category
```

Comment:
Remove `category` field: this is a association so it belong to `_embedded` field